release_config_test.go

```
@@ -0,0 +1,46 @@
+	if strings.Contains(goreleaser, "go mod tidy") {
+		t.Fatal(".goreleaser.yaml must not run go mod tidy during a release")
+	}
+
+	workflow := releaseConfigFile(t, filepath.Join(root, ".github", "workflows", "release.yml"))
+	setupGo := strings.Index(workflow, "- name: Set up Go")
+	tidyCheck := strings.Index(workflow, "run: go mod tidy -diff")
+	goreleaserStep := strings.Index(workflow, "- name: Run GoReleaser")
+	if setupGo == -1 || tidyCheck == -1 || goreleaserStep == -1 {
+		t.Fatal("release workflow must set up Go, check tidy module metadata, and run GoReleaser")
+	}
+	if !(setupGo < tidyCheck && tidyCheck < goreleaserStep) {
```

golangci-lint (2.13.2)

[error] QF1001: could apply De Morgan's law

(staticcheck)